ab_tool/canvas.py

```diff
@@ -92,12 +90,41 @@ def _get_installed_intervention_point_urls(self):
         return installed_intervention_point_urls
 
 
-def get_unsorted_students(request, experiment):
+def get_unassigned_students(request, experiment):
+    """ Gets the list of students that have not been assigned a track.
+        This version of the function is for requests that have come through
+        a web view and consequently have a django request object off of which
+        to make a canvas request context """
     request_context = get_canvas_request_context(request)
-    course_id = get_lti_param(request, "custom_canvas_course_id")
+    return get_unassigned_students_with_context(request_context, experiment)
+
+
+def get_unassigned_students_with_stored_credentials(course_object, experiment):
```

Comment:
It seems unnecessary to me to pass in both the experiment and the course_object and have the below check.  Where this function is called in the management command, the experiment is being derived from the course, so why not just pass in the experiment?  You should be able to specify in the models module that an Experiment belongs to a Course and make an experiment.course call by modifying the model.